### ISSUE TITLE
vmagent panic

### ISSUE THREAD

**Issue by cocotyty** [VictoriaMetrics/VictoriaMetrics#964]
**Describe the bug**
It would be great [upgrading](https://victoriametrics.github.io/#how-to-upgrade) to [the latest avaialble release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases)
and verifying whether the bug is reproducible there.
It is also recommended reading [troubleshooting docs](https://victoriametrics.github.io/#troubleshooting).
vmagent cannot work.
```text
2020-12-14T12:00:18.492Z	panic	VictoriaMetrics/lib/persistentqueue/persistentqueue.go:495	BUG: readerOffset=2286868 cannot exceed writerOffset=1230778
panic: BUG: readerOffset=2286868 cannot exceed writerOffset=1230778

goroutine 131 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage(0xb267de, 0x5, 0xc0067e0900, 0x3c, 0x4)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:258 +0xc25
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, 0xb267de, 0x5, 0xb483e7, 0x32, 0xc000721d78, 0x2, 0x2)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:126 +0xd1
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:118
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:114
github.com/VictoriaMetrics/VictoriaMetrics/lib/persistentqueue.(*Queue).MustReadBlock(0xc0002e0000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/persistentqueue/persistentqueue.go:495 +0x356
github.com/VictoriaMetrics/VictoriaMetrics/lib/persistentqueue.(*FastQueue).MustReadBlock(0xc00029ef50, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/persistentqueue/fastqueue.go:169 +0x149
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*client).runWorker(0xc00017c380)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/client.go:161 +0xa6
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.newClient.func1(0xc00017c380)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/client.go:124 +0x54
created by github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.newClient
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/client.go:122 +0x85f
```
**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**

**Screenshots**

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ ./victoria-metrics-prod --version
victoria-metrics-20190730-121249-heads-single-node-0-g671d9e55
```
v1.49.0
**Used command-line flags**
Command-line flags are listed as `flag{name="httpListenAddr", value=":443"} 1` lines at `/metrics` page.
See the following docs for details:

* [monitoring for single-node VictoriaMetrics](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#monitoring)
* [montioring for VictoriaMetrics cluster](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cluster/README.md#monitoring)

**Additional context**
Add any other context about the problem here such as error logs from VictoriaMetrics and Prometheus,
`/metrics` output, screenshots from the official Grafana dashboards for VictoriaMetrics:

* [Grafana dashboard for single-node VictoriaMetrics](https://grafana.com/dashboards/10229)
* [Grafana dashboard for VictoriaMetrics cluster](https://grafana.com/grafana/dashboards/11176)

**Comment by valyala** [comment]
The bug must be fixed in the commit d006b41eff60da036af2623c123ccd587f31d323 . This commit will be included in the next release. Temporary workaround is to remove the contents of directory pointed by `-remoteWrite.tmpDataPath` command-line flag before starting `vmagent`. If this flag isn't set, then `vmagent-remotewrite-data` directory is used.

**Comment by cocotyty** [comment]
thank you! it works.

**Comment by valyala** [comment]
The bugfix has been included in [v1.50.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
